libs/ledger-live-common/src/families/vechain/bridge.integration.test.ts

```diff
@@ -0,0 +1,212 @@
+                recipient: "0x17733CAb76d9A2112576443F21735789733B1ca3",
+                amount: "20000000000000000000",
+                body: {
+                  chainTag: MAINNET_CHAIN_TAG,
+                  blockRef: "0x00634a0c856ec1db",
+                  expiration: 18,
+                  clauses: [
+                    {
+                      to: "0x17733CAb76d9A2112576443F21735789733B1ca3",
+                      value: "20000000000000000000",
+                      data: "0x",
+                    },
+                  ],
+                  gasPriceCoef: DEFAULT_GAS_COEFFICIENT,
+                  gas: "0",
+                  dependsOn: null,
+                  nonce: generateNonce(),
+                },
+              }),
+              expectedStatus: {
+                amount: new BigNumber("20000000000000000000"),
+                errors: {
+                  amount: new NotEnoughBalance(),
+                },
+                warnings: {},
+                totalSpent: new BigNumber("0"),
+                estimatedFees: new BigNumber("0"),
+              },
+            },
+            {
+              name: "VTHO balance not enough",
+              transaction: fromTransactionRaw({
+                family: "vechain",
+                subAccountId:
+                  "js:2:vechain:0x82d984c42fC9d49E2d4CFFdcad59301AfFca7E02:vechain+vechain%2Fvtho",
+                estimatedFees: new BigNumber(0),
+                recipient: "0x17733CAb76d9A2112576443F21735789733B1ca3",
+                amount: "20000000000000000000",
+                body: {
+                  chainTag: MAINNET_CHAIN_TAG,
+                  blockRef: "0x00634a0c856ec1db",
+                  expiration: 18,
+                  clauses: [
+                    {
+                      to: "0x17733CAb76d9A2112576443F21735789733B1ca3",
+                      value: "20000000000000000000",
+                      data: "0x",
+                    },
+                  ],
+                  gasPriceCoef: DEFAULT_GAS_COEFFICIENT,
+                  gas: "0",
+                  dependsOn: null,
+                  nonce: generateNonce(),
+                },
+              }),
+              expectedStatus: {
+                amount: new BigNumber("20000000000000000000"),
+                errors: {
+                  amount: new NotEnoughBalance(),
+                },
+                warnings: {},
+                totalSpent: new BigNumber("0"),
+                estimatedFees: new BigNumber("0"),
+              },
+            },
+          ],
+          FIXME_tests: [
+            "balance is sum of ops", //the balance depends on VTHO and it's earned without operations
+            "empty transaction is equals to itself", //nonce is not deterministic
+            "ref stability on self transaction", //blockref is not deterministic
+            "can be run in parallel and all yield same results", //blockref is not deterministic
```

Comment:
For both, you can now use `IgnorePrepareTransactionFields` instead

Reply:
sorry, what do you mean by that exactly?

Reply:
btw I tried with uncommenting them and they seem to work without adding the IgnorePrepareTransactionFields, so probably it was an old issue

Reply:
no sorry we have the problem on ubuntu

Reply:
I'll add them back

Reply:
My proposal was to use [this](https://github.com/LedgerHQ/ledger-live/blob/173d7d6d224bcf1cecf364062b6571f52792e371/libs/ledgerjs/packages/types-live/src/bridge.ts#L232).
You can find an example [here](https://github.com/LedgerHQ/ledger-live/blob/173d7d6d224bcf1cecf364062b6571f52792e371/libs/coin-evm/src/__tests__/integration/bridge.integration.test.ts#L16).

You can use it for both `nonce` and `blockref`. Doing so will just ignore those fields but still run the tests, it's less restrictive than disabling the tests.